frontera/contrib/messagebus/zeromq/socket_config.py

```diff
@@ -1,29 +1,62 @@
 # -*- coding: utf-8 -*-
+"""
+Contains the SocketConfig class
+"""
+from socket import getaddrinfo, gaierror
 
 
 class SocketConfig(object):
-
-    hostname = None
-    base_port = None
-
-    def __init__(self, hostname, base_port):
-        self.hostname = hostname
-        self.base_port = base_port
+    """
+    Converts address to IPv4 or IPv6 or * and returns the necessary socket
+    addresses.
+    NOTE: When using * it defaults to IPv4
```

Comment:
Man, if you were able to bind the socket on Travis CI on IPv6 address, then we have everything to make testing IPv6 work.
`ZMQ_ADDRESS` setting can be used with wildcard, it could be useful when running on different machines using config.
No, I won't surrender :) I'm going to find a machine in Scrapinghub, or setup a new one to test it in IPv6 environment.
If you don't want to continue, I'll take your code and finish it, just tell me.